File: check_coc_units.js
/**
 * COC API Data Checker - Check all material quantities and units
 */

const axios = require('axios');

const COC_API_URL = 'https://umanmrp.in/api/coc_api.php';

async function checkCOCUnits() {
    try {
        console.log('🔍 Fetching COC data from API...\n');
        
        // Get last 6 months data
        const toDate = new Date().toISOString().split('T')[0];
        const fromDate = new Date(Date.now() - 180*24*60*60*1000).toISOString().split('T')[0];
        
        const response = await axios.post(COC_API_URL, {
            from: fromDate,
            to: toDate
        }, { timeout: 30000 });
        
        let cocData = [];
        if (response.data && response.data.data) {
            cocData = response.data.data;
        } else if (Array.isArray(response.data)) {
            cocData = response.data;
        }
        
        console.log(`✅ Fetched ${cocData.length} COC records\n`);
        console.log('='*80);
        console.log('📊 MATERIAL WISE QUANTITY ANALYSIS\n');
        console.log('='*80);
        
        // Group by material name
        const materialGroups = {};
        
        cocData.forEach(coc => {
            const material = coc.material_name || 'Unknown';
            
            if (!materialGroups[material]) {
                materialGroups[material] = {
                    count: 0,
                    quantities: [],
                    units: new Set(),
                    samples: []
                };
            }
            
            materialGroups[material].count++;
            
            // Store quantity info
            if (coc.coc_qty) {
                materialGroups[material].quantities.push(parseFloat(coc.coc_qty) || 0);
            }
            
            // Try to detect unit from quantity field or product name
            if (coc.product_name) {
                materialGroups[material].units.add(coc.product_name);
            }
            
            // Store sample data
            if (materialGroups[material].samples.length < 3) {
                materialGroups[material].samples.push({
                    invoice: coc.invoice_no,
                    qty: coc.coc_qty,
                    invoice_qty: coc.invoice_qty,
                    product: coc.product_name,
                    brand: coc.brand
                });
            }
        });
        
        // Print material-wise summary
        const sortedMaterials = Object.keys(materialGroups).sort();
        
        sortedMaterials.forEach(material => {
            const data = materialGroups[material];
            const avgQty = data.quantities.reduce((a, b) => a + b, 0) / data.quantities.length;
            const minQty = Math.min(...data.quantities);
            const maxQty = Math.max(...data.quantities);
            
            console.log(`\n📦 ${material}`);
            console.log(`   Records: ${data.count}`);
            console.log(`   Qty Range: ${minQty.toFixed(2)} - ${maxQty.toFixed(2)} (Avg: ${avgQty.toFixed(2)})`);
            
            console.log(`   Sample Entries:`);
            data.samples.forEach((sample, idx) => {
                console.log(`      ${idx + 1}. Invoice: ${sample.invoice}, COC Qty: ${sample.qty}, Invoice Qty: ${sample.invoice_qty}`);
                console.log(`         Product: ${sample.product || 'N/A'}, Brand: ${sample.brand || 'N/A'}`);
            });
        });
        
        console.log('\n' + '='*80);
        console.log('✅ Analysis Complete!');
        console.log('='*80);
        
    } catch (error) {
        console.error('❌ Error:', error.message);
        if (error.response) {
            console.error('Response:', error.response.data);
        }
    }
}

// Run the checker
checkCOCUnits();

File: check_coc_units.py
"""
COC API Data Checker - Check all material quantities and units
"""

import requests
from datetime import datetime, timedelta
from collections import defaultdict
import json

COC_API_URL = 'https://umanmrp.in/api/coc_api.php'

def check_coc_units():
    try:
        print('🔍 Fetching COC data from API...\n')
        
        # Get last 6 months data
        to_date = datetime.now().strftime('%Y-%m-%d')
        from_date = (datetime.now() - timedelta(days=180)).strftime('%Y-%m-%d')
        
        response = requests.post(COC_API_URL, json={
            'from': from_date,
            'to': to_date
        }, timeout=30)
        
        coc_data = []
        if response.status_code == 200:
            data = response.json()
            if isinstance(data, dict) and 'data' in data:
                coc_data = data['data']
            elif isinstance(data, list):
                coc_data = data
        
        print(f'✅ Fetched {len(coc_data)} COC records\n')
        print('='*80)
        print('📊 MATERIAL WISE QUANTITY ANALYSIS\n')
        print('='*80)
        
        # Group by material name
        material_groups = defaultdict(lambda: {
            'count': 0,
            'quantities': [],
            'units': set(),
            'samples': []
        })
        
        for coc in coc_data:
            material = coc.get('material_name', 'Unknown')
            
            material_groups[material]['count'] += 1
            
            # Store quantity info
            if coc.get('coc_qty'):
                try:
                    qty = float(coc['coc_qty'])
                    material_groups[material]['quantities'].append(qty)
                except:
                    pass
            
            # Try to detect unit from product name
            if coc.get('product_name'):
                material_groups[material]['units'].add(coc['product_name'])
            
            # Store sample data (max 3 samples per material)
            if len(material_groups[material]['samples']) < 3:
                material_groups[material]['samples'].append({
                    'invoice': coc.get('invoice_no', 'N/A'),
                    'qty': coc.get('coc_qty', 'N/A'),
                    'invoice_qty': coc.get('invoice_qty', 'N/A'),
                    'product': coc.get('product_name', 'N/A'),
                    'brand': coc.get('brand', 'N/A')
                })
        
        # Print material-wise summary
        sorted_materials = sorted(material_groups.keys())
        
        for material in sorted_materials:
            data = material_groups[material]
            
            if data['quantities']:
                avg_qty = sum(data['quantities']) / len(data['quantities'])
                min_qty = min(data['quantities'])
                max_qty = max(data['quantities'])
                
                print(f'\n📦 {material}')
                print(f'   Records: {data["count"]}')
                print(f'   Qty Range: {min_qty:.2f} - {max_qty:.2f} (Avg: {avg_qty:.2f})')
                
                print(f'   Sample Entries:')
                for idx, sample in enumerate(data['samples'], 1):
                    print(f'      {idx}. Invoice: {sample["invoice"]}, COC Qty: {sample["qty"]}, Invoice Qty: {sample["invoice_qty"]}')
                    print(f'         Product: {sample["product"]}, Brand: {sample["brand"]}')
        
        print('\n' + '='*80)
        print('✅ Analysis Complete!')
        print('='*80)
        
    except Exception as error:
        print(f'❌ Error: {str(error)}')

if __name__ == '__main__':
    check_coc_units()
